client/sites-dashboard/components/p2-thumbnail.tsx

```diff
@@ -0,0 +1,126 @@
+import styled from '@emotion/styled';
+import { useI18n } from '@wordpress/react-i18n';
+import type { SitesDisplayMode } from './sites-display-mode-switcher';
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+const LARGE_ICON_PX = 64;
+const SMALL_ICON_PX = 32;
+const ICON_BORDER_PX = 3;
+
+const headerImageSize = {
+	width: '100%',
+	height: 'auto',
+	aspectRatio: '5/2',
+};
+
+const Container = styled.div( {
+	userSelect: 'none',
+	position: 'relative',
+	top: 0,
+	left: 0,
+	width: '100%',
+	height: '100%',
+	overflow: 'hidden',
+
+	// Using flex to remove an unexplained gap between the header image and site icon
+	display: 'flex',
+	flexDirection: 'column',
+	alignItems: 'start',
+} );
+
+const HeaderImage = styled.img( {
+	...headerImageSize,
+	objectFit: 'cover',
+} );
+
+const ColorGradient = styled.div( {
+	...headerImageSize,
+
+	'::before': {
+		content: '""',
+		display: 'block',
+		width: '100%',
+		height: '100%',
+		background: 'linear-gradient( 80.15deg, rgba( 0, 0, 0, .3 ) 0%, rgba( 0, 0, 0, 0 ) 100% )',
+	},
+} );
+
+const SiteIconContainer = styled.div< { isSmall: boolean } >(
+	{
+		background: '#fff',
+		padding: ICON_BORDER_PX,
+		borderRadius: '4px',
+		position: 'relative',
+	},
+	( { isSmall } ) =>
+		isSmall
+			? {
+					left: '8px',
+					top: `${ -SMALL_ICON_PX / 2 - ICON_BORDER_PX }px`,
+			  }
+			: {
+					left: '32px',
+					top: `${ -LARGE_ICON_PX / 2 - ICON_BORDER_PX }px`,
+			  }
+);
+
+const SiteIcon = styled.img< { isSmall: boolean } >(
+	{
+		borderRadius: '2px',
+		display: 'block',
+	},
+	( { isSmall } ) =>
+		isSmall
+			? {
+					width: SMALL_ICON_PX,
+					height: SMALL_ICON_PX,
+			  }
+			: {
+					width: LARGE_ICON_PX,
+					height: LARGE_ICON_PX,
+			  }
+);
+
+interface P2ThumbnailProps {
+	site: SiteExcerptData;
+	displayMode: SitesDisplayMode;
+}
+
+export function P2Thumbnail( { site, displayMode }: P2ThumbnailProps ) {
+	const { __ } = useI18n();
+
+	if ( ! site.p2_thumbnail_elements ) {
+		return null;
+	}
+
+	const { color_link, color_sidebar_background, header_image } = site.p2_thumbnail_elements;
+
+	function renderContents() {
+		const isSmall = displayMode === 'list';
+
+		return (
+			<>
+				{ header_image ? (
+					<HeaderImage src={ header_image } alt="" />
```

Comment:
Should we have some default alt text for `<HeaderImage />` (and `<SiteIcon />` below)?

Also, I think we'd want `width=` and `height=` attributes on them as well.

Lastly, the site icon seems a little blurry, and the header image is quite large:

<img width="869" alt="image" src="https://user-images.githubusercontent.com/36432/195833480-4bf58b8c-a092-4feb-8661-42136e327cb9.png">

Given we can manipulate image sizes using query strings, it'd probably be good to do some `sizes` and `srcset` action.

Reply:
> Should we have some default alt text for `<HeaderImage />` (and `<SiteIcon />` below)?

I don't think they should have separate alt text, but I do think the alt text situation can be improved. The parent div has `role=img`, so my hope is that it's treated as a single image. This seems to be the case when I test it in VoiceOver. But "Site Icon" probably isn't the right alt text. I'm changing it to `site.title || 'Site thumbnail'` since I think it should be treated the same as the actual mShot screenshot.

> Also, I think we'd want width= and height= attributes on them as well.

We know the size of the icon so I can do that. The rendered size of the header image is trickier though, so I'll need to do the same `srcset` tricks that we do for mShots I think.

> I was a little concerned the different icon placement would look odd, but I think it looks fine.

Me too. If the dashboard was filled with only private sites and P2s then I think it'd feel a bit off. But in the context of a more realistic dashboard all the thumbnails look a little different so I think the inconsistency is "fine"

Reply:
@danielbachhuber I've added `srcset` and `sizes` for the header image and site icon. Mercifully I don't need to calculate sizes for the site icon since it's a fixed size, and we only need to specify retina and non-retina. I'm not sure it's worth including a `3x` image. Is it likely users will have uploaded an image with that much resolution? Doesn't seem worth it.

I had originally calculated a custom `sizes` attribute for the header image, until I realised I could reuse the `sizes` attribute from the mShot images since they're the same width 🤦 
The sizes I manually came up with were a little different from those already committed, so I updated them to match what I'd come up with. I think the existing ones had been generated using a browser extension is that right? I feel like mine are more accurate 😎 

In terms of `srcset` for the header image, I feel like the most benefit comes when you actually know what image sizes we have on the server. Obviously we could specify hundreds of images, each with their own `?w` param, but that doesn't seem like the right thing to do. So I've only specified two image resolutions which are based on the suggested image size in the P2 customizer.

<img width="301" alt="CleanShot 2022-10-17 at 18 27 42@2x" src="https://user-images.githubusercontent.com/1500769/196096079-ebf7c1a2-e81d-48a3-b602-4a14d9d298b4.png">

That seems like our best guess at what size is actually on the server.